integration-tests/src/test/java/oracle/kubernetes/operator/ItJrfModelInImage.java

```diff
@@ -0,0 +1,243 @@
+      
+      Secret walletPass = assertDoesNotThrow(() -> new WalletPasswordSecret(namespace, 
+          domainUid + "-opss-wallet-password-secret", walletPassword));
+      LoggerHelper.getLocal().log(Level.INFO, "Wallet password secret is created for {0}", domainUid);
+      
+      jrfdomain = assertDoesNotThrow(() -> new JrfDomain(domainMap), 
+          "Failed: JRF domain creation");
+      LoggerHelper.getLocal().log(Level.INFO, "JRF domain is created for {0}", testClassName);
+      assertDoesNotThrow(() -> jrfdomain.verifyDomainCreated(40), 
+          "Failed: domain verification");
+      LoggerHelper.getLocal().log(Level.INFO, "JRF domain verification succeeded for {0}", testClassName);
+      
+      //save and restore walletFile secret
+      assertDoesNotThrow(() -> TestUtils.saveWalletFileSecret(getResultDir(), domainUid, namespace),
+          "Failed to save walletFile secret");
+      LoggerHelper.getLocal().log(Level.INFO, "Saved walletFile secret for {0}", domainUid);
+      String walletFileSecretName = domainUid + "-opss-walletfile-secret";
+      assertDoesNotThrow(() -> TestUtils.restoreWalletFileSecret(getResultDir(), domainUid, namespace, 
+          walletFileSecretName), 
+          "Failed to store walletFile secret");
+      LoggerHelper.getLocal().log(Level.INFO, "Restored walletFile secret for {0}", domainUid);
+      
+      //shutdown the domain
+      assertDoesNotThrow(() -> jrfdomain.shutdownUsingServerStartPolicy(),
+          "Failed to shutdown the domain");
+      LoggerHelper.getLocal().log(Level.INFO, "Shutdown the domain {0}", domainUid);
+      
+      //modify the original domain to enable walletFileSecret
+      String originalYaml = getUserProjectsDir() + "/weblogic-domains/" + jrfdomain.getDomainUid()
+          + "/domain.yaml"; 
+      DomainCrd crd = assertDoesNotThrow(() -> new DomainCrd(originalYaml));
+      Map<String, String> opssNode = new HashMap();
+      opssNode.put("walletFileSecret", walletFileSecretName);
+      crd.addObjectNodeToOpss(opssNode);
+      String modYaml = assertDoesNotThrow(() -> crd.getYamlTree());
+      LoggerHelper.getLocal().log(Level.INFO, modYaml);
+      // Write the modified yaml to a new file
+      Path path = Paths.get(getUserProjectsDir() + "/weblogic-domains/" + jrfdomain.getDomainUid(),
+          "modified.domain.yaml");
+      LoggerHelper.getLocal().log(Level.INFO, "Path of the modified domain.yaml :{0}", path.toString());
+      Charset charset = StandardCharsets.UTF_8;
+      assertDoesNotThrow(() -> Files.write(path, modYaml.getBytes(charset)));
+      
+      //Use the new yaml to startup the domain
+      LoggerHelper.getLocal().log(Level.INFO, "kubectl apply -f {0}", path.toString());
+      ExecResult exec = assertDoesNotThrow(() -> TestUtils.exec("kubectl apply -f " + path.toString()));
+      LoggerHelper.getLocal().log(Level.INFO, exec.stdout());
+      
+      assertDoesNotThrow(() -> jrfdomain.verifyDomainCreated(40));
```

Comment:
is this enough? don't you need to check the domain has the right jdbc credentials? how do you know if the change worked or not?

Reply:
For this test case checking restarted domain that reuses the same RCU schema that is used when the domain initially created is in the good state,  meaning the required pods, services are created and the servers are ready should be enough.
There is no JDBC functionality involved here.
If the DB credentials are wrong RCU schema can not be created.